app/Http/Controllers/AdminController.php

```diff
@@ -1484,4 +1485,116 @@ private function gitinfo_file_contents(string $filename): string
 
         return $html;
     }
+
+    /**
+     * TODO: (williamjallen) Rewrite this to take advantage of Blade templating.
+     */
+    public function generateCTestConfig(): View|RedirectResponse|\Illuminate\Http\Response
+    {
+        @set_time_limit(0);
+
+        @$projectid = $_GET['projectid'];
+
+        // Checks
+        if (!isset($projectid) || !is_numeric($projectid)) {
+            return view('cdash', [
+                'xsl' => true,
+                'xsl_content' => 'Not a valid projectid!',
+                'title' => 'CTest Configuration'
+            ]);
+        }
+
+        $db = Database::getInstance();
+        $project_array = $db->executePreparedSingleRow('SELECT * FROM project WHERE id=?', [intval($projectid)]);
+        if (empty($project_array)) {
+            return view('cdash', [
+                'xsl' => true,
+                'xsl_content' => 'Not a valid projectid!',
+                'title' => 'CTest Configuration'
+            ]);
+        }
+        $policy = checkUserPolicy($project_array['id']);
+
+        if ($policy !== true) {
+            return $policy;
+        }
+
+        $ctestconfig = "## This file should be placed in the root directory of your project.\n";
+        $ctestconfig .= "## Then modify the CMakeLists.txt file in the root directory of your\n";
+        $ctestconfig .= "## project to incorporate the testing dashboard.\n";
+        $ctestconfig .= "##\n";
+        $ctestconfig .= "## # The following are required to submit to the CDash dashboard:\n";
+        $ctestconfig .= "##   ENABLE_TESTING()\n";
+        $ctestconfig .= "##   INCLUDE(CTest)\n";
+        $ctestconfig .= "\n";
+
+        $ctestconfig .= 'set(CTEST_PROJECT_NAME "' . $project_array['name'] . "\")\n";
+        $ctestconfig .= 'set(CTEST_NIGHTLY_START_TIME "' . $project_array['nightlytime'] . "\")\n\n";
+
+        if (config('app.env') === 'production') {
+            $ctestconfig .= "set(CTEST_DROP_METHOD \"https\")\n";
+        } else {
+            $ctestconfig .= "set(CTEST_DROP_METHOD \"http\")\n";
+        }
+
+        $currentURI = str_replace('https://', '', env('APP_URL'));
+        $currentURI = str_replace('http://', '', $currentURI);
+
+        $ctestconfig .= 'set(CTEST_DROP_SITE "' . $currentURI . "\")\n";
+
+        $ctestconfig .= 'set(CTEST_DROP_LOCATION "' . $currentURI . '/submit.php?project=' . urlencode($project_array['name']) . "\")\n";
```

Comment:
`DROP_LOCATION` shouldn't have the URI as a part of the string. It should only be the `submit.php` plus the project name.

CTest will mash them all together at submission time